src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/RenamerProjectTreeActionHandler.HandleFileMove.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Runtime.Remoting.Contexts;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Rename;
+using Microsoft.VisualStudio.ProjectSystem.Waiting;
+using Microsoft.VisualStudio.Settings;
+using Solution = Microsoft.CodeAnalysis.Solution;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Rename
+{
+    internal partial class RenamerProjectTreeActionHandler
+    {
+        private readonly HashSet<Renamer.RenameDocumentActionSet> _results = new HashSet<Renamer.RenameDocumentActionSet>();
+
+        protected virtual async Task CPSCopyAsync(IProjectTreeActionHandlerContext context, IReadOnlyList<IProjectTree> items, DropEffects effect)
+        {
+            await base.CopyAsync(context, items, effect);
+        }
+
+        public override async Task CopyAsync(IProjectTreeActionHandlerContext context, IReadOnlyList<IProjectTree> items, DropEffects effect)
+        {
+            Requires.NotNull(context, nameof(Context));
+            Requires.NotNull(items, nameof(items));
+
+            _results.Clear();
+
+            ISettingsManager settings = await _settingsManagerService.GetValueAsync();
+            bool EnableNamespaceUpdate = settings.GetValueOrDefault("SolutionNavigator.EnableNamespaceUpdate", false);
+
+            if (EnableNamespaceUpdate)
+            {
+                CodeAnalysis.Project? project = GetCurrentProject();
+
+                if (project is null)
+                {
+                    return;
+                }
+
+                foreach (var item in items)
+                {
+                    string oldFilePath = ""; // Get filename
+                    string filenameWithExtension = ""; // remote extension
+                    string destinationFolder = ""; // Get destination folder name;
+
+                    (bool result, Renamer.RenameDocumentActionSet? documentActions)
+                        = await GetRenameSymbolsActions(project, oldFilePath, filenameWithExtension, new List<string>(new string[] { destinationFolder }));
+
+                    if (result == true)
```

Comment:
```suggestion
                    if (result)
```

Consider renaming `result` to something more specific such as `success`.